packages/typespec-ts/src/modular/buildOperations.ts

```diff
@@ -25,6 +26,7 @@ export function buildOperationFiles(
   needUnexpectedHelper: boolean = true
 ) {
   for (const operationGroup of client.operationGroups) {
+    utilImports.clear();
```

Comment:
it feels odd to have a module-level Set that gets cleared by this method. Does the set need to live outside this method or could it be created when buildOperationFiles is called?

Reply:
I put this outside the scope of this buildOperationFiles functions as explained here  https://github.com/Azure/autorest.typescript/pull/1934/files#r1282500160 

I need to clear this set because the apis per each operationGroup should be generated in an independent file.

Reply:
oh I see. I'm not wholly against it, but it does create some complexity that could cause unexpected behavior later if someone isn't careful.

Maybe we could later refactor this a bit - I remember when I wrote a prototype TS emitter I had a global bag of context that I passed through every generation method that handled things like tracking deps, etc.

Reply:
Thanks for the suggestion, will take that into consideration when I try to adopt to TCGC.